### PR TITLE
UI Changes, General Improvements & Bug Fixes

### PR DESCRIPTION
**Pull request by OmniacDev**
**UI Changes:**
- Added UI themes: Light, Dark, and System.
- Added custom titlebar with window buttons.

**Dark Theme:**
Screenshot 2024-06-25 233030

**Light Theme:**
Screenshot 2024-06-25 233041

**Theme Switching Demo**
https://www.youtube.com/watch?v=jmpc7DD3XLY

**Other:**
- Pruned some unused npm packages
- Added "dev" npm script to make it easier to debug (instead of having to manually run "start-react" and "start-electron")
- Fixed some other bugs

### CODE REVIEW

**Comment by theaddonn** [comment]
Need to say, I prefer the old side bar icons...

**Comment by FrederoxDev** [comment]
When you're ready for this to be reviewed let me know. Could you also add images of any new UI changes? Thanks

**Comment by OmniacDev** [comment]
@FrederoxDev done

**Comment by FrederoxDev** [comment]
Could you attach some images of all the changes?

**Comment by OmniacDev** [comment]
Check original

**Comment by OmniacDev** [comment]
up the top

**Comment by FrederoxDev** [comment]
Cool, do you think you could make the theme selector a dropdown similar to the profile editor dropdown for the runtimes. I really like the new settings icon, looks a lot cleaner.

**Comment by OmniacDev** [comment]
gonna be changing it to 3 buttons for light dark and system, but first I needa make everything else follow the theme

**Comment by FrederoxDev** [comment]
I think only the title bar should really follow that. As mentioned in the discord server, adding other themes is just adding more to have to maintain, and id rather keep that as small as possible for future updates

**Comment by OmniacDev** [comment]
it's actually really easy to modify

**Comment by OmniacDev** [comment]
check out the `titlebar.css` file you'll see

**Comment by FrederoxDev** [comment]
Realistically in a months time, the launcher is probably going to get left to me to maintain again, and I'd prefer to be spending as little time as possible working on the launcher, it already takes me forever to do anything with one theme xD. plus I'd rather stay out of CSS and in tailwind

**Comment by OmniacDev** [comment]
I mean I'll still do it anyway. Also what is there really to maintain? The launcher seems to already have everything it needs, I can't really think of anything that could be added that would be a problem

**Comment by FrederoxDev** [comment]
Eventually there will still need to be a few more panels. Mod config editors, proper version management, and a few other things. 

As a developer, I need to choose where and what I want to put my time towards, and the launcher really isn't that much of a priority to myself so I choose to spend as little time as possible on it. 

I think changing the color of the titlebar is good enough right now. Plus I think it looks good in both themes.

**Comment by Tallis-Larsen** [comment]
This looks good! The settings icon looks a lot better than I thought it would. I was doubtful, but I'm impressed!

**Comment by OmniacDev** [comment]
Added the button panel for the themes.

ready to merge

**Comment by FrederoxDev** [comment]
Could you attach some screenshots of the new radial buttons?

File: src/pages/SettingsPage.tsx [hunk 1]
```diff
@@ -60,22 +62,80 @@ export default function SettingsPage() {
...
+            </div>
+
             <ToggleSection
                 text="Developer mode"
                 subtext="Enables hot-reloading and prompting to attach a debugger."
                 isChecked={developerMode}
                 setIsChecked={setDeveloperMode}
             />
+
+            <div className="border-y-[2px] border-solid border-b-[#1E1E1F] border-t-[#5A5B5C] p-[8px] bg-[#48494A]">
```
Comment (FrederoxDev) [inline src/pages/SettingsPage.tsx]: This entire section for the radial buttons should be made generic and placed into its own component

**Review by FrederoxDev — CHANGES_REQUESTED** [review]

**Comment by OmniacDev** [comment]
https://github.com/FrederoxDev/Amethyst-Launcher/assets/64994204/c5635972-c3f7-4f6f-a518-69a2e311245a

File: src/pages/SettingsPage.tsx (hunk 1 above)
Comment (OmniacDev) [reply src/pages/SettingsPage.tsx]: done

**Comment by FrederoxDev** [comment]
Cool, looks good to me. One very minor thing is could you add a white title to that section similar to all of the rest. After that probably good to merge

**Comment by OmniacDev** [comment]
image

**Comment by OmniacDev** [comment]
final

**Review by FrederoxDev — APPROVED** [review]

**Pull request merged**